r/R/expression.R

```diff
@@ -56,13 +56,9 @@ build_array_expression <- function(FUN,
                                    args = list(...),
                                    options = empty_named_list()) {
   if (FUN == "-" && length(args) == 1L) {
-    # Unary -, i.e. just make it negative, and somehow this works
     if (inherits(args[[1]], c("ArrowObject", "array_expression"))) {
-      # Make it be 0 - arg
-      # TODO(ARROW-11950): do this in C++ compute
-      args <- list(0L, args[[1]])
+      return(build_array_expression("negate", args[[1]]))
```

Comment:
Question: do we want to use "negate" or "negate_checked" here? https://github.com/apache/arrow/blob/master/cpp/src/arrow/compute/kernels/scalar_arithmetic.cc#L514-L524

We have tended to use the "checked" versions of the other arithmetic functions, but we haven't thought too hard about it.

Reply:
I think the edge case which `negate_checked` watches for is probably not necessary here. @edponce

Reply:
Perhaps for the sake of consistency, using `negate_checked` makes sense, even if not strictly necessary?

Reply:
Alright, note that `negate_checked` does not support unsigned arguments:
```r
call_function("negate_checked", args=list(Array$create(c(1,2), type=uint8())))
#> Error: NotImplemented: Function negate_checked has no kernel matching input types (array[uint8])
```

Reply:
I just noticed that we should update the *negate_checked_doc* to state explicitly that it does not support unsigned arguments. We do show this in the function signature table https://github.com/apache/arrow/blob/master/docs/source/cpp/compute.rst#arithmetic-functions

Reply:
I would think that using the *checked* versions for language bindings is more desirable as you want errors to be triggered ASAP. The *unchecked* versions should be used knowing the wrap around behavior may occur. I think of it as *unchecked = safe math* and *checked = true math*.

Reply:
I just noticed that we should update the negate_checked_doc to state explicitly that it does not support unsigned arguments. We do show this in the function signature table https://github.com/apache/arrow/blob/master/docs/source/cpp/compute.rst#arithmetic-functions

Reply:
I would think that using the checked versions for language bindings is more desirable as you want errors to be triggered ASAP. The unchecked versions should be used knowing the wrap around behavior may occur. I think of it as unchecked = safe math and checked = true math